src/test/java/uk/gov/companieshouse/officerfiling/api/controller/ValidationStatusControllerImplTest.java

```diff
@@ -1,57 +1,141 @@
+import uk.gov.companieshouse.officerfiling.api.service.CompanyAppointmentService;
+import uk.gov.companieshouse.officerfiling.api.service.CompanyProfileService;
 import uk.gov.companieshouse.officerfiling.api.service.OfficerFilingService;
-import uk.gov.companieshouse.officerfiling.api.utils.LogHelper;
+import uk.gov.companieshouse.officerfiling.api.service.TransactionService;
+import uk.gov.companieshouse.sdk.manager.ApiSdkManager;
 
 @ExtendWith(MockitoExtension.class)
 class ValidationStatusControllerImplTest {
 
-    public static final String TRANS_ID = "117524-754816-491724";
-    public static final String FILING_ID = "6332aa6ed28ad2333c3a520a";
+    private static final String TRANS_ID = "117524-754816-491724";
+    private static final String FILING_ID = "6332aa6ed28ad2333c3a520a";
+    private static final String ETAG = "etag";
+    private static final String COMPANY_TYPE = "ltd";
+    private static final String COMPANY_NUMBER = "COMPANY_NUMBER";
+    private static final String PASSTHROUGH_HEADER = "passthrough";
     @Mock
     private OfficerFilingService officerFilingService;
     @Mock
     private HttpServletRequest request;
     @Mock
     private Logger logger;
     @Mock
-    private LogHelper logHelper;
-
+    private TransactionService transactionService;
+    @Mock
+    private CompanyProfileService companyProfileService;
+    @Mock
+    private CompanyAppointmentService companyAppointmentService;
+    @Mock
+    private OfficerFilingMapper officerFilingMapper;
+    @Mock
+    private Transaction transaction;
+    @Mock
+    private AppointmentFullRecordAPI companyAppointment;
+    @Mock
+    private CompanyProfileApi companyProfile;
+    @Mock
+    private OfficerFilingDto dto;
+    @InjectMocks
+    private ErrorMapper errorMapper = new ErrorMapperImpl();
+    private OfficerFiling filing;
     private ValidationStatusControllerImpl testController;
 
+
     @BeforeEach
     void setUp() {
-        testController = new ValidationStatusControllerImpl(officerFilingService, logger);
+        testController = new ValidationStatusControllerImpl(officerFilingService, logger,
+            transactionService, companyProfileService, companyAppointmentService, officerFilingMapper,
+            errorMapper);
+        filing = OfficerFiling.builder()
+            .referenceAppointmentId("off-id")
+            .referenceEtag("etag")
+            .resignedOn(Instant.parse("2022-09-13T00:00:00Z"))
+            .build();
+
     }
 
     @Test
-    void validateWhenFound() {
-        var filing = OfficerFiling.builder().build();
+    void validateWhenFilingNotFound() {
+        when(officerFilingService.get(FILING_ID, TRANS_ID)).thenReturn(Optional.empty());
+        when(transaction.getId()).thenReturn(TRANS_ID);
+
+        assertThrows(ResourceNotFoundException.class, () -> testController.validate(transaction, FILING_ID, request));
+    }
+    @Test
+    void validateWhenFilingFoundAndNoValidationErrors() {
```

Comment:
Some repetition of `when()` stubbings here and in `validateWhenFilingFoundAndValidationErrors()` could be extracted to method.

Reply:
Now changed to suggested implementation.

Reply:
Now changed to suggested implementation.